rollout/canary.go

```diff
@@ -292,17 +297,40 @@ func (c *RolloutController) syncRolloutStatusCanary(roCtx *canaryContext) error
 	}
 
 	currBackgroundAr := analysisutil.GetCurrentBackgroundAnalysisRun(currArs)
-	if currBackgroundAr != nil {
-		if currBackgroundAr.Status == nil || !currBackgroundAr.Status.Status.Completed() || analysisutil.IsTerminating(currBackgroundAr) {
+	if currBackgroundAr != nil && !r.Status.Abort {
+		if currBackgroundAr.Status == nil || !currBackgroundAr.Status.Status.Completed() {
```

Comment:
I'm not sure why I made AnalysisRun.Status a bool pointer. Seems like it just makes the code annoying to deal with.

Reply:
Yeah, it adds a lot of nil checks.